core/src/com/unciv/ui/tilegroups/CityButton.kt

```diff
@@ -31,25 +30,334 @@ import com.unciv.ui.utils.extensions.colorFromRGB
 import com.unciv.ui.utils.extensions.onClick
 import com.unciv.ui.utils.extensions.toGroup
 import com.unciv.ui.utils.extensions.toLabel
-import javax.swing.Icon
 import kotlin.math.max
 import kotlin.math.min
 
-private object Colors {
-    val construction = colorFromRGB(196,140,62)
-    val growh =  colorFromRGB(130,225,78)
+class InfluenceTable(
+    influence: Float,
+    relationshipLevel: RelationshipLevel,
+    width: Float=100f,
+    height: Float=5f) : Table() {
+
+    override fun draw(batch: Batch?, parentAlpha: Float) { super.draw(batch, parentAlpha) }
+
+    init {
+
+        defaults().pad(1f)
+        setSize(width, height)
+
+        val normalizedInfluence = max(-60f, min(influence, 60f)) / 30f
+
+        val color = when (relationshipLevel) {
+            RelationshipLevel.Unforgivable -> Color.RED
+            RelationshipLevel.Enemy -> Color.ORANGE
+            RelationshipLevel.Afraid -> Color.YELLOW
+            RelationshipLevel.Neutral, RelationshipLevel.Friend -> Color.LIME
+            RelationshipLevel.Ally -> Color.SKY
+            else -> Color.DARK_GRAY
```

Comment:
Looks like `DiplomacyManager.RelationshipLevel` already has these colors coded so they might be worth reusing.

Reply:
Those colors are different, they are used for diplomacy/leader screen.
E.g. `Ally` color for diplomacy is green, while `Ally` indicator for influence bar should be blue.